Gems/GradientSignal/Code/Source/UI/GradientPreviewWidget.h

```diff
@@ -8,26 +8,40 @@
 
 #pragma once
 
+#if !defined(Q_MOC_RUN)
 #include <QWidget>
 
 #include <GradientSignal/Editor/EditorGradientPreviewRenderer.h>
+#endif
+
+class QToolButton;
 
 namespace GradientSignal
 {
     class GradientPreviewWidget
         : public QWidget
         , public EditorGradientPreviewRenderer
     {
+        Q_OBJECT
+
     public:
-        GradientPreviewWidget(QWidget* parent = nullptr);
+        GradientPreviewWidget(bool enablePopout = false, QWidget* parent = nullptr);
```

Comment:
nit: usually in Qt constructors I'd expect the parent to be first and other params to come after it (especially since they all have defaults). Not sure if it's a fixed rule or just my expectation though.

Reply:
Yeah we have it mixed around in our code-base, but in general I agree, I will swap!